### PR TITLE
Adds ignorecase field to pageable spring

### PR DESCRIPTION
**Pull request by gklp**
Adds ignorecase field to pageable spring for compatibility with SortHandlerMethodArgumentResolverSupport::parseIgnoreCase

### CODE REVIEW

File: spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/encoding/FeignPageableEncodingTests.java [hunk 1]
```diff
@@ -264,6 +265,110 @@ void testSortWithBody() {
...
+		assertThat(response.getBody()).isNotNull();
+		assertThat(pageable.getPageSize()).isEqualTo(response.getBody().getSize());
+		assertThat(response.getBody().getPageable().getSort()).hasSize(1);
+		Optional<Sort.Order> optionalOrder = response.getBody().getPageable().getSort().get().findFirst();
+		assertThat(optionalOrder.isPresent()).isEqualTo(true);
+		if (optionalOrder.isPresent()) {
+			Sort.Order order = optionalOrder.get();
+			assertThat(order.getDirection()).isEqualTo(Sort.Direction.ASC);
+			assertThat(order.getProperty()).isEqualTo("anySorting");
+			assertThat(order.isIgnoreCase()).as("isIgnoreCase has not expected value.").isEqualTo(true);
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/encoding/FeignPageableEncodingTests.java:289]: Please rephrase the message to "isIgnoreCase does not have expected value"

File: spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/encoding/FeignPageableEncodingTests.java [hunk 2]
```diff
@@ -264,6 +265,110 @@ void testSortWithBody() {
...
+		Sort sort = response.getBody().getPageable().getSort();
+		assertThat(sort).hasSize(2);
+
+		List<Sort.Order> orderList = sort.toList();
+		assertThat(orderList).hasSize(2);
+
+		Sort.Order firstOrder = orderList.get(0);
+		assertThat(firstOrder.getDirection()).isEqualTo(Sort.Direction.DESC);
+		assertThat(firstOrder.getProperty()).isEqualTo("anySorting1");
+		assertThat(firstOrder.isIgnoreCase()).as("isIgnoreCase has not expected value.").isEqualTo(true);
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/encoding/FeignPageableEncodingTests.java:362]: Please rephrase the message to "isIgnoreCase does not have expected value"

File: spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/encoding/FeignPageableEncodingTests.java [hunk 3]
```diff
@@ -264,6 +265,110 @@ void testSortWithBody() {
...
+		assertThat(firstOrder.getNullHandling()).isEqualTo(Sort.NullHandling.NATIVE);
+
+		Sort.Order secondOrder = orderList.get(1);
+		assertThat(secondOrder.getDirection()).isEqualTo(Sort.Direction.ASC);
+		assertThat(secondOrder.getProperty()).isEqualTo("anySorting2");
+		assertThat(secondOrder.isIgnoreCase()).as("isIgnoreCase has not expected value.").isEqualTo(false);
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/encoding/FeignPageableEncodingTests.java:368]: Please rephrase the message to "isIgnoreCase does not have expected value"

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Hi, @gklp. Thanks for submitting the PR. Have added some cosmetic comments - please address them. Also, please update the date in the license comments of any files you've modified to `2013-2024` .

**Comment by gklp** [comment]
all issues were resolved @OlgaMaciaszek

File: spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/encoding/FeignPageableEncodingTests.java [hunk 4]
```diff
@@ -264,6 +265,110 @@ void testSortWithBody() {
...
+		assertThat(response).isNotNull();
+		assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+		assertThat(response.getBody()).isNotNull();
+		assertThat(sort).isEqualTo(response.getBody().getSort());
+
+		Optional<Sort.Order> optionalOrder = response.getBody().getPageable().getSort().get().findFirst();
+		assertThat(optionalOrder.isPresent()).isEqualTo(true);
+		if (optionalOrder.isPresent()) {
+			Sort.Order order = optionalOrder.get();
+			assertThat(order.isIgnoreCase()).as("isIgnoreCase has not expected value.").isEqualTo(true);
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/encoding/FeignPageableEncodingTests.java:314]: Please rephrase the message to "isIgnoreCase does not have expected value"

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Hi @gklp, it still needs one minor change - please see the comment.

File: spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/encoding/FeignPageableEncodingTests.java (hunk 4 above)
Comment (gklp) [reply spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/encoding/FeignPageableEncodingTests.java]: All "isIgnoreCase has not expected value." descriptions were changed.

**Review by OlgaMaciaszek — APPROVED** [review]
Thanks @gklp. LGTM.

**Pull request merged**